spec/integ/matrix-client-event-timeline.spec.ts

```diff
@@ -52,12 +53,16 @@ const withoutRoomId = (e: Partial<IEvent>): Partial<IEvent> => {
 /**
  * Our httpBackend only allows matching calls if we have the exact same query, in the exact same order
  * This method allows building queries with the exact same parameter order as the fetchRelations method in client
+ * @param client Matrix client to mock the request for
  * @param params query parameters
  */
-const buildRelationPaginationQuery = (params: QueryDict): string => {
+const buildRelationPaginationQuery = (client: MatrixClient, params: QueryDict): string => {
     if (Thread.hasServerSideFwdPaginationSupport === FeatureSupport.Experimental) {
         params = replaceParam("dir", "org.matrix.msc3715.dir", params);
     }
+    if (client.canSupport.get(Feature.RelationsRecursion) === ServerSupport.Unstable) {
+        params = replaceParam("recurse", "org.matrix.msc3981.recurse", params);
+    }
```

Comment:
this feels like a sledgehammer to crack a nut. Shouldn't the calling test know if we expect `recurse` or `org.matrix.msc3981.recurse`?